### PR TITLE
feat: add exception types

### PR DESCRIPTION
**Pull request by baszalmstra**
Add exception types to python bindings

### CODE REVIEW

**Comment by baszalmstra** [comment]
Isnt that the case? There is an `exceptions.py` that contains all the exceptions. Or do you think we shouldn't reexport them?

File: py-rattler/rattler/__init__.py [hunk 1]
```diff
@@ -16,11 +16,43 @@
...
+    InvalidMatchSpecError,
+    InvalidPackageNameError,
+    InvalidUrlError,
+    InvalidVersionError,
+    IoError,
+    LinkError,
+    ParseArchError,
+    ParsePlatformError,
+    TransactionError,
+)
```
Comment (Wackyator) [inline py-rattler/rattler/__init__.py:34]:
```suggestion
    SolverError,
)
```

File: py-rattler/rattler/__init__.py [hunk 2]
```diff
@@ -16,11 +16,43 @@
...
+    "FetchRepoDataError",
+    "InvalidChannelError",
+    "InvalidMatchSpecError",
+    "InvalidPackageNameError",
+    "InvalidUrlError",
+    "InvalidVersionError",
+    "IoError",
+    "LinkError",
+    "ParseArchError",
+    "ParsePlatformError",
```
Comment (Wackyator) [inline py-rattler/rattler/__init__.py:53]:
```suggestion
    "ParsePlatformError",
    "SolverError",
```

File: py-rattler/rattler/exception.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        InvalidUrlError,
+        InvalidChannelError,
+        ActivationError,
+        ParsePlatformError,
+        ParseArchError,
+        TransactionError,
+        LinkError,
+        IoError,
+        DetectVirtualPackageError,
+        CacheDirError,
```
Comment (Wackyator) [inline py-rattler/rattler/exception.py:15]:
```suggestion
        CacheDirError,
        SolverError,
```

File: py-rattler/rattler/exception.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        """An error that can occur during io operations"""
+
+    class DetectVirtualPackageError(Exception):  # type: ignore[no-redef]
+        """An error that can occur when trying to detect virtual packages"""
+
+    class CacheDirError(Exception):  # type: ignore[no-redef]
+        """An error that can occur when querying the cache directory"""
+
+    class FetchRepoDataError(Exception):  # type: ignore[no-redef]
+        """An error that can occur when fetching repo data"""
```
Comment (Wackyator) [inline py-rattler/rattler/exception.py:62]:
```suggestion
        """An error that can occur when fetching repo data"""
        
    class SolverError(Exception): #type: ignore[no-redef]
        """An error that can occur when solving dependencies"""
```

File: py-rattler/rattler/exception.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    "InvalidChannelError",
+    "InvalidMatchSpecError",
+    "InvalidPackageNameError",
+    "InvalidUrlError",
+    "InvalidVersionError",
+    "IoError",
+    "LinkError",
+    "ParseArchError",
+    "ParsePlatformError",
+    "TransactionError",
```
Comment (Wackyator) [inline py-rattler/rattler/exception.py:79]:
```suggestion
    "TransactionError",
    "SolverError",
```

**Comment by Wackyator** [comment]
We exposed the `SolverError` in ffi lib but never exposed in Python

**Comment by baszalmstra** [comment]
I added `SolverError` and removed the reexport!

**Comment by baszalmstra** [comment]
Should we also add them to the docs? @Wackyator what would be a good approach?

**Review by Wackyator — APPROVED** [review]

**Comment by baszalmstra** [comment]
Cool! Just add it to this PR if you want!

**Review by wolfv — APPROVED** [review]

**Pull request merged**